### ISSUE TITLE
The Source release artifact should only have a folder on the first level

### ISSUE THREAD

**Issue by freijon** [helix-editor/helix#6586]
Hello

The idiomatic way to create and distribute tarballs is to have them unpack into a folder which contains everything (like the archive files from Github). Otherwise when you unpack the tarball we have files 'all over the place'. Obviously this is nothing critical, but a small workaround is needed in the Gentoo ebuild to account for this.

Ideally the source gets unpacked to a folder called `helix-<version>.tar.xz`.

**Comment by dead10ck** [comment]
Apologies, but "idiomatic" in what sense? It's pretty standard for large Rust projects to be split up into crates. Even many C projects I've seen in the past eventually find a need to organize their files into directories. The only projects that can reasonably have all their source files in one flat directory are small projects; otherwise, it gets very hard to maintain.

**Comment by dead10ck** [comment]
Ohh I see, sorry I totally misinterpreted what you said, thanks for the clarification!

**Comment by archseer** [comment]
Isn't `mkdir src && tar -xf helix.tar.xz -C src` enough to work around this?

**Comment by archseer** [comment]
GNU tar also supports `--one-top-level`/`--one-top-level=new_directory`

**Comment by freijon** [comment]
All these workarounds you mention do work indeed if you unpack the archive manually. Portage by default just unpacks the archive, and assumes the source is in a folder called <name>-<version>, which is true is most cases (like Github archives). There are workarounds for this as well, that's why I mention it's nothing critical.
Is it a big deal for you to change this? Is there a specific reason you want to keep it this way?

**Comment by the-mikedavis** [comment]
It would be a breaking change for any existing packagers that use the `helix-<version>-source.tar.xz`

**Comment by freijon** [comment]
Indeed. I propose this change for the next release, existing archives should not be touched. Packagers will have to account for this modification when they package the next release. But I'm certain this change will be a welcome one for packagers who use the source archive. This is the common way to distribute tarballs.

**Comment by ljmccarthy** [comment]
Unfortunately this issue hit me and I lost data. I extracted the tarball in my home directory, expecting it to create a single directory, but it overwrote my .envrc file which I use to set all my environment variables. I also had to carefully clean up all the files splatted all over my home directory... Yes I'm an idiot, but expectations matter.
